<!DOCTYPE html>



  


<html class="theme-next muse use-motion" lang="zh-Hans">
<head>
  <meta charset="UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>
<meta name="theme-color" content="#222">









<meta http-equiv="Cache-Control" content="no-transform" />
<meta http-equiv="Cache-Control" content="no-siteapp" />
















  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css" />







<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css" />

<link href="/css/main.css?v=5.1.4" rel="stylesheet" type="text/css" />


  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png?v=5.1.4">


  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png?v=5.1.4">


  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png?v=5.1.4">


  <link rel="mask-icon" href="/images/logo.svg?v=5.1.4" color="#222">





  <meta name="keywords" content="Hexo, NexT" />










<meta name="description" content="find a way">
<meta property="og:type" content="website">
<meta property="og:title" content="find my way">
<meta property="og:url" content="http://yoursite.com/page/2/index.html">
<meta property="og:site_name" content="find my way">
<meta property="og:description" content="find a way">
<meta property="og:locale" content="zh-Hans">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="find my way">
<meta name="twitter:description" content="find a way">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Muse',
    version: '5.1.4',
    sidebar: {"position":"left","display":"post","offset":12,"b2t":false,"scrollpercent":false,"onmobile":false},
    fancybox: true,
    tabs: true,
    motion: {"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},
    duoshuo: {
      userId: '0',
      author: '博主'
    },
    algolia: {
      applicationID: '',
      apiKey: '',
      indexName: '',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="http://yoursite.com/page/2/"/>





  <title>find my way</title>
  








</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-Hans">

  
  
    
  

  <div class="container sidebar-position-left 
  page-home">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <div class="custom-logo-site-title">
      <a href="/"  class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">find my way</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <p class="site-subtitle">accumulate</p>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br />
            
            首页
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br />
            
            归档
          </a>
        </li>
      

      
    </ul>
  

  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            
  <section id="posts" class="posts-expand">
    
      

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://yoursite.com/2020/03/19/Median-Joining Networks for Inferring Intraspecific Phylogenies/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="HaolinZhong">
      <meta itemprop="description" content="">
      <meta itemprop="image" content="https://timgsa.baidu.com/timg?image&quality=80&size=b9999_10000&sec=1534345547319&di=b9e880681b19e0cefb3132f38ccb0281&imgtype=0&src=http%3A%2F%2Fimg.duoziwang.com%2F2017%2F09%2F1607450267817.jpg">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="find my way">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
                
                <a class="post-title-link" href="/2020/03/19/Median-Joining Networks for Inferring Intraspecific Phylogenies/" itemprop="url">未命名</a></h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2020-03-19T19:52:03+08:00">
                2020-03-19
              </time>
            

            

            
          </span>

          

          
            
          

          
          

          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p>Median-Joining Networks for Inferring Intraspecific Phylogenies</p>
<p>软件NETWORK使用的算法</p>
<h3 id="Median-Joining"><a href="#Median-Joining" class="headerlink" title="Median Joining"></a>Median Joining</h3><h4 id="Prerequisites"><a href="#Prerequisites" class="headerlink" title="Prerequisites"></a>Prerequisites</h4><p>input data：correctly  aligned  sequences  of  a  population  sample.（确保ambiguous  states  are  infrequent  and recombination is absent.）</p>
<h4 id="Distances"><a href="#Distances" class="headerlink" title="Distances"></a>Distances</h4><p>最简单的度量距离方法：数两条序列之间差异的character个数(the ‘‘Hamming distance’’).优化：给character changes加上权重；</p>
<h4 id="Ambiguous-States"><a href="#Ambiguous-States" class="headerlink" title="Ambiguous States"></a>Ambiguous States</h4><p>例如X=R（purine），意味着X可能是{A, G}; X=N意味着X可能是{A,G,C,T}。计算这些含有多种可能状态的距离时，取最小距离；</p>
<h4 id="The-algorithm"><a href="#The-algorithm" class="headerlink" title="The algorithm"></a>The algorithm</h4><p>parameter：Increasing the parameter $\varepsilon $ widens the search for potential new median vectors </p>
<p>Step1:distance matrix</p>
<p><img src="/Users/zhonghaolin/Hexo/source/_posts/image/median_joining_table_1.png" alt=""></p>
<p>$\delta_{1}=4, \delta_{2}=5, \delta_{3}=6, \delta_{4}=7$</p>
<p>Step2、3: </p>
<p>$\delta=\delta_{1}=4$ 有{A,B,C} and {D} </p>
<p>$\delta=\delta_{2}=5$ 有{A,B,C,D}</p>
<p>Step4:Compute the minimum connection cost $\lambda$ for all feasible triplets U, V, and W.</p>
<p>Triplets and median vector </p>
<p><img src="/Users/zhonghaolin/Hexo/source/_posts/image/median-joining_figure2.png" alt=""></p>
<p>Thus, A, B, C and A, B, D constitute feasible triplets,from which median vectors U = 10000 and V = 01000 can be generated at connection costs 7 and 8.</p>
<p>Phase II: Construction of the final network</p>
<p>Step 5:</p>
<table>
<thead>
<tr>
<th></th>
<th>A</th>
<th>B</th>
<th>C</th>
<th>D</th>
<th>U</th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td>A</td>
<td>*</td>
<td>4</td>
<td>4</td>
<td>7</td>
<td>1</td>
<td></td>
</tr>
<tr>
<td>B</td>
<td></td>
<td>*</td>
<td>6</td>
<td>5</td>
<td>3</td>
<td></td>
</tr>
<tr>
<td>C</td>
<td></td>
<td></td>
<td>*</td>
<td>7</td>
<td>3</td>
<td></td>
</tr>
<tr>
<td>D</td>
<td></td>
<td></td>
<td></td>
<td>*</td>
<td>8</td>
<td></td>
</tr>
<tr>
<td>U</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>*</td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<p>ref：<a href="https://wenku.baidu.com/view/fce316eab8f67c1cfad6b80f.html" target="_blank" rel="noopener">系统发育的网络构建与应用</a></p>
<p><strong>To be continued</strong></p>

          
        
      
    </div>
    
    
    

    

    

    

    <footer class="post-footer">
      

      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </div>
  
  
  
  </article>


    
      

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://yoursite.com/2020/03/16/linux服务器安装tensorflow配置GPU/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="HaolinZhong">
      <meta itemprop="description" content="">
      <meta itemprop="image" content="https://timgsa.baidu.com/timg?image&quality=80&size=b9999_10000&sec=1534345547319&di=b9e880681b19e0cefb3132f38ccb0281&imgtype=0&src=http%3A%2F%2Fimg.duoziwang.com%2F2017%2F09%2F1607450267817.jpg">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="find my way">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
                
                <a class="post-title-link" href="/2020/03/16/linux服务器安装tensorflow配置GPU/" itemprop="url">未命名</a></h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2020-03-16T16:16:52+08:00">
                2020-03-16
              </time>
            

            

            
          </span>

          

          
            
          

          
          

          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p>[TOC]</p>
<p>tensorflow官网：<a href="https://tensorflow.google.cn/" target="_blank" rel="noopener">https://tensorflow.google.cn/</a></p>
<p>安装教程：<a href="https://blog.csdn.net/qq_38901147/article/details/90049666" target="_blank" rel="noopener">https://blog.csdn.net/qq_38901147/article/details/90049666</a></p>
<p><img src="/Users/zhonghaolin/Hexo/source/_posts/image/tensorflow_gpu_cuda.png" alt="tensorflow_cuda_version"></p>
<p>环境：linux+Anaconda3+python3.5+CUDA9.0+cudnn7.0</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span> cuda9.0 for tensorflow1.6</span><br><span class="line">export PATH="/home/haolinz/CUDA/cuda-9.0/bin:PATH" </span><br><span class="line">export LD_LIBRARY_PATH="/home/haolinz/CUDA/cuda-9.0/lib64:$LD_LIBRARY_PATH" </span><br><span class="line">export CUDA_HOME=/home/haolinz/CUDA/cuda-9.0</span><br></pre></td></tr></table></figure>
<p>###1.conda 创建虚拟环境(python3.6+tensorflow1.6)</p>
<p><code>conda create -n tensorflow1.6 python=3.6</code></p>
<p>安装tensorflow1.6(cpu版本)</p>
<p><code>pip install tensorflow==1.6</code></p>
<p>###2.配置GPU</p>
<p>####2.1Nvidia驱动（服务器已经安装）</p>
<p>安装后输入<code>nvidia-smi</code>会显示显卡信息；</p>
<p><img src="/Users/zhonghaolin/Hexo/source/_posts/image/nvidia_smi.png" alt=""></p>
<p>####2.2 CUDA下载及安装</p>
<p>查看linux内核，对于isyslab服务器g-1-4，g-1-5是centos 7.3 redhat，g-1-6是6.5；    </p>
<p><code>cat /etc/redhat-release</code></p>
<p>确认CUDA与cudnn配对版本,这里选择tensorflow1.6对应的cuDNN7，CUDA9；（最新的tensorflow需要使用CUDA10.1）</p>
<p><img src="/Users/zhonghaolin/Hexo/source/_posts/image/tensorflow_cuda_cudnn.png" alt=""></p>
<p>下载CUDA9.0如下：</p>
<p><img src="/Users/zhonghaolin/Hexo/source/_posts/image/CUDA_download.png" alt=""></p>
<p>输入<code>sh cuda_9.0.176_384.81_linux-run</code>进行安装</p>
<p>按如下输入：</p>
<p><img src="/Users/zhonghaolin/Hexo/source/_posts/image/cuda安装步骤.png" alt=""></p>
<p>安装后完成后：export环境变量</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span> cuda9.0 for tensorflow1.6</span><br><span class="line">export PATH=/home/haolinz/CUDA/cuda-9.0/bin$&#123;PATH:+:$&#123;PATH&#125;&#125;</span><br><span class="line">export LD_LIBRARY_PATH=/home/haolinz/CUDA/cuda-9.0/lib64$&#123;LD_LIBRARY_PATH:+:$&#123;LD_LIBRARY_PATH&#125;&#125;</span><br><span class="line">export CUDA_HOME=/home/haolinz/CUDA/cuda-9.0</span><br></pre></td></tr></table></figure>
<p><strong>服务器已经安装了cuda，直接导入就行了</strong></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">export PATH=/usr/local/cuda/bin$&#123;PATH:+:$&#123;PATH&#125;&#125;</span><br><span class="line">export LD_LIBRARY_PATH=/usr/local/cuda/lib64$&#123;LD_LIBRARY_PATH:+:$&#123;LD_LIBRARY_PATH&#125;&#125;</span><br><span class="line">export CUDA_HOME=/usr/local/cuda</span><br></pre></td></tr></table></figure>
<p>然后输入<code>nvcc -V</code>，如果显示CUDA版本信息代表正常。</p>
<p>测试sample例子：最后都是PASS说明CUDA安装成功；</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span>编译并测试设备 deviceQuery：</span><br><span class="line">make</span><br><span class="line">cd ~/CUDA/cuda-9.0/samples/1_Utilities/deviceQuery</span><br><span class="line">./deviceQuery</span><br><span class="line"><span class="meta">#</span>编译并测试带宽 bandwidthTest：</span><br><span class="line">make</span><br><span class="line">cd ../bandwidthTest</span><br><span class="line">./bandwidthTest</span><br></pre></td></tr></table></figure>
<p>####2.3 cuDNN下载及配置</p>
<p>到<a href="https://developer.nvidia.com/rdp/cudnn-archive" target="_blank" rel="noopener">nvidia网站</a>下载与CUDA9.0对应的cuDNN</p>
<p>教程里是下载得到tgz文件直接解压，我用mac下载是.arj文件，服务器上没有arj解压（或者我不知道），最后是在mac上先解压cudnn-9.0-linux-x64-v7.6.5.32.arj得到cudnn-9.0-linux-x64-v7.6.5.32，这个文件还是个压缩文件，猜测是tgz文件，于是<code>tar -zxvf cudnn-9.0-linux-x64-v7.6.5.32</code>,还真的解压出了，把解压缩得到的文件上传到服务器，拷贝到相应位置</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span> 将库(.lib)和头文件(.h)copy到cuda目录（一定是你自己安装的目录如/mnt/jackfrost/CUDA/cuda-9.0）</span><br><span class="line">cp ~/tensorflow/cuDNN/cuda/include/cudnn.h ~/CUDA/cuda-9.0/include/</span><br><span class="line">cp ~/tensorflow/cuDNN/cuda/lib64/libcudnn* ~/CUDA/cuda-9.0/lib64/</span><br><span class="line"><span class="meta">#</span>修改文件访问权限：</span><br><span class="line">chmod a+r ~/CUDA/cuda-9.0/include/cudnn.h ~/CUDA/cuda-9.0/lib64/libcudnn*</span><br></pre></td></tr></table></figure>
<p>###3.pycharm远程连接服务器</p>
<p>利用frp实现内网穿透，frp github项目：<a href="https://github.com/fatedier/frp" target="_blank" rel="noopener">https://github.com/fatedier/frp</a></p>
<p>博客实操说明：<a href="https://cloud.tencent.com/developer/article/1536406" target="_blank" rel="noopener">https://cloud.tencent.com/developer/article/1536406</a></p>
<p>我有两个服务器一个是实验室的服务器isyslab.info,一个是芋头的华为云119.3.148.222</p>
<p>3.1 Modify <code>frps.ini</code> on server A（华为云）:</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span> frps.ini</span><br><span class="line">[common]</span><br><span class="line">bind_port = 7000</span><br></pre></td></tr></table></figure>
<p>3.2 Start <code>frps</code> on server A（华为云）:</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">./frps -c ./frps.ini    # 最好是让其在服务器退出后还能运行，使用nohup或是tmux</span><br></pre></td></tr></table></figure>
<p>3.3 On server B（实验室）, modify <code>frpc.ini</code> to put in your <code>frps</code> server public IP as <code>server_addr</code> field:</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span> frpc.ini</span><br><span class="line">[common]</span><br><span class="line">server_addr = 119.3.148.222</span><br><span class="line">server_port = 7000</span><br><span class="line"></span><br><span class="line">[ssh]</span><br><span class="line">type = tcp</span><br><span class="line">local_ip = 127.0.0.1</span><br><span class="line">local_port = 22</span><br><span class="line">remote_port = 6000</span><br></pre></td></tr></table></figure>
<p>3.4 Start <code>frpc</code> on server B:</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">./frpc -c ./frpc.ini    # 使用nohup或是tmux后台运行</span><br></pre></td></tr></table></figure>
<h3 id="4-小结"><a href="#4-小结" class="headerlink" title="4.小结"></a>4.小结</h3><p>1.在isyslab的gpu节点上，各个gpu节点的配置是不同的：例如g-1-4，g-1-5是centos 7.3 redhat，g-1-6是6.5。一开始在g-1-6安装tensorflow1.6报错”libc.so.6: version ‘GLIBC_2.14’ not found”，查看系统glibc支持的版本：<code>strings /lib64/libc.so.6 |grep GLIBC_</code>,发现确实没有，但是在g-1-4和g-1-5上都有，所以在这两个节点上安装成功。</p>
<p>2.使用conda创建虚拟环境和使用virtualenv创建虚拟环境(<a href="https://blog.csdn.net/lyy14011305/article/details/59500819" target="_blank" rel="noopener">ref1</a> ,)：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span> 1.查看当前存在哪些虚拟环境</span><br><span class="line">conda env list    # 或 </span><br><span class="line">conda info -e	</span><br><span class="line"><span class="meta">#</span>2.检查更新当前conda</span><br><span class="line">conda update conda</span><br></pre></td></tr></table></figure>
<p>3.tmux使用简单说明（<a href="https://www.cnblogs.com/kevingrace/p/6496899.html" target="_blank" rel="noopener">ref1</a> ,）</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p>4.pycharm连接服务器(<a href="https://blog.csdn.net/baidu_39622935/article/details/84299439" target="_blank" rel="noopener">ref1</a>)</p>

          
        
      
    </div>
    
    
    

    

    

    

    <footer class="post-footer">
      

      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </div>
  
  
  
  </article>


    
      

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://yoursite.com/2020/03/12/Decoding the evolution and transmissions of the novel pneumonia coronavirus (SARS CoV 2) using whole genomic data/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="HaolinZhong">
      <meta itemprop="description" content="">
      <meta itemprop="image" content="https://timgsa.baidu.com/timg?image&quality=80&size=b9999_10000&sec=1534345547319&di=b9e880681b19e0cefb3132f38ccb0281&imgtype=0&src=http%3A%2F%2Fimg.duoziwang.com%2F2017%2F09%2F1607450267817.jpg">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="find my way">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
                
                <a class="post-title-link" href="/2020/03/12/Decoding the evolution and transmissions of the novel pneumonia coronavirus (SARS CoV 2) using whole genomic data/" itemprop="url">未命名</a></h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2020-03-12T19:51:34+08:00">
                2020-03-12
              </time>
            

            

            
          </span>

          

          
            
          

          
          

          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p> <strong>Decoding the evolution and transmissions of the novel pneumonia coronavirus (SARS-CoV-2) using whole genomic data</strong> </p>
<p>Ref:<a href="https://www.thepaper.cn/newsDetail_forward_6079509,http://blog.sciencenet.cn/blog-2277-1219899.html" target="_blank" rel="noopener">https://www.thepaper.cn/newsDetail_forward_6079509,http://blog.sciencenet.cn/blog-2277-1219899.html</a></p>
<p>###Background</p>
<p>In this study, we used 93 complete genomes of SARS CoV 2 from the GISAID EpiCoVTM database to investigate the evolution and human to human transmissions of SARS CoV 2 in the recent two months.</p>
<p>###Methods</p>
<p>####1.数据下载</p>
<p><a href="https://www.gisaid.org" target="_blank" rel="noopener">GISAID EpiCoV</a>数据库下载93个新冠病毒基因组加上1个中华菊头蝠中的冠状病毒基因组;</p>
<p>####2.基因组比对</p>
<p>使用MAFFT对基因组进行序列比对，所有fasta序列需要合并到一个文件作为输入文件，直接使用<code>cat *fasta &gt; input</code>有问题，fasta之间没有换行符；（使用网上的一个脚本或是自己写一个）</p>
<p>命令行运行<code>mafft total.fasta &gt; output.txt</code> 得到比对文件output.txt</p>
<h4 id="3-geneious观察比对结果"><a href="#3-geneious观察比对结果" class="headerlink" title="3.geneious观察比对结果"></a>3.geneious观察比对结果</h4><p><img src="/Users/zhonghaolin/Library/Application Support/typora-user-images/image-20200312211747568.png" alt="image-20200312211747568"></p>
<p>coding regions：</p>
<p><img src="/Users/zhonghaolin/Library/Application Support/typora-user-images/image-20200312212426333.png" alt="image-20200312212426333"></p>
<h4 id="3-DnaSP（ref1，ref2）"><a href="#3-DnaSP（ref1，ref2）" class="headerlink" title="3.DnaSP（ref1，ref2）"></a>3.DnaSP（<a href="https://www.plob.org/article/3682.html" target="_blank" rel="noopener">ref1</a>，<a href="https://www.plob.org/article/3841.html" target="_blank" rel="noopener">ref2</a>）</h4><p>序列文件里有些字符DnaSP软件不能识别：核苷酸链中除出现正常的A、T、G、C四种碱基符号外还出现如R、Y、M、K等其它字母，这些字母表示有几种可能性。（其中R=A/G，Y=C/T，M=A/C，K=G/T，S=C/G，W= A/T，H= A/C/T，B= C/G/T，V=A/C/G， D=A/G /T,N=A/C/G/T）</p>
<p>EPI_ISL_407071, EPI_ISL_407894, EPI_ISL_407896, and EPI_ISL_409067 这四个基因组的特殊字符需要用常用ATCG代替；</p>
<p>多序列比对结果使用DnaSP分析，1.单倍型分析，去除蝙蝠默认生成文件得到59种单倍型，文章58种；</p>
<p><img src="/Users/zhonghaolin/Library/Application Support/typora-user-images/image-20200312215454792.png" alt="image-20200312215454792"></p>
<p>2.同义突变与非同义突变的分析；</p>
<h4 id="4-NetWORK绘制单倍型网络"><a href="#4-NetWORK绘制单倍型网络" class="headerlink" title="4.NetWORK绘制单倍型网络"></a>4.NetWORK绘制单倍型网络</h4><p>ref：<a href="http://blog.sciencenet.cn/blog-3152652-1155330.html" target="_blank" rel="noopener">http://blog.sciencenet.cn/blog-3152652-1155330.html</a></p>
<p>DnaSP生成相应单倍型文件，输入到NetWORK中生成网络</p>
<p>94个基因型未排列</p>
<p><img src="/Users/zhonghaolin/Hexo/source/_posts/image/test_94_network.bmp" alt=""></p>
<p>按照原文排列（不含bat）</p>
<p><img src="/Users/zhonghaolin/Hexo/source/_posts/image/origianl_haplotype_93_network.bmp" alt=""></p>
<h4 id="5-IQ-TREE"><a href="#5-IQ-TREE" class="headerlink" title="5.IQ-TREE"></a>5.IQ-TREE</h4><p>IQ-TREE使用网页版，依据Haplotype构建进化树，使用figTree可视化并且修改</p>
<p><img src="/Users/zhonghaolin/Hexo/source/_posts/image/tree.png" alt=""></p>
<h4 id="6-population-size"><a href="#6-population-size" class="headerlink" title="6.population size"></a>6.population size</h4><p>ref：<a href="http://blog.sciencenet.cn/blog-460481-900027.html" target="_blank" rel="noopener">http://blog.sciencenet.cn/blog-460481-900027.html</a></p>
<p>使用DnaSNP分析的种群population size(mismatch distribution)</p>
<p><img src="/Users/zhonghaolin/Hexo/source/_posts/image/mismatch_distribution.jpg" alt=""></p>
<p>pass</p>

          
        
      
    </div>
    
    
    

    

    

    

    <footer class="post-footer">
      

      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </div>
  
  
  
  </article>


    
      

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://yoursite.com/2020/03/01/machine_learning/Machine_learning_李宏毅2019/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="HaolinZhong">
      <meta itemprop="description" content="">
      <meta itemprop="image" content="https://timgsa.baidu.com/timg?image&quality=80&size=b9999_10000&sec=1534345547319&di=b9e880681b19e0cefb3132f38ccb0281&imgtype=0&src=http%3A%2F%2Fimg.duoziwang.com%2F2017%2F09%2F1607450267817.jpg">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="find my way">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
                
                <a class="post-title-link" href="/2020/03/01/machine_learning/Machine_learning_李宏毅2019/" itemprop="url">未命名</a></h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2020-03-01T21:54:35+08:00">
                2020-03-01
              </time>
            

            

            
          </span>

          

          
            
          

          
          

          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p>[TOC]</p>
<p>###introduction</p>
<p>Bilibili地址：<a href="https://www.bilibili.com/video/av73798762?p=24" target="_blank" rel="noopener">2019李宏毅机器学习入门</a></p>
<h3 id="W1"><a href="#W1" class="headerlink" title="W1"></a>W1</h3><h4 id="1-机器学习的介绍"><a href="#1-机器学习的介绍" class="headerlink" title="1.机器学习的介绍"></a>1.机器学习的介绍</h4><p>​    pass</p>
<h4 id="2-regression-case-study"><a href="#2-regression-case-study" class="headerlink" title="2.regression-case study"></a>2.regression-case study</h4><p>回归模型，主要介绍的是最简单的线性回归模型，y=b+wx，预测精灵宝可梦进化后的战力值变化，模型复杂从一元一次变为二次，三次，甚至四次并且在这个过程中介绍过拟合现象（<strong>overfitting</strong>）；及模型好坏的评价指标损失函数（<strong>loss function</strong>）：最简单的损失函数相当于概率论里的平方差即（预测值-实际值）平方，稍微复杂一点的是加上w的权重使选出的模型更加平滑（smooth）；使用梯度下降（<strong>gradient descent</strong>）来寻找最优解，大致公式为w0-（待定）；</p>
<p>step 1：Model，选择模型；</p>
<p>step 2: goodness of function，衡量模型好坏的函数；</p>
<p>step 3：Best function（best gradient），寻找模型的最佳参数即调参；</p>
<h4 id="3-gradient-descent"><a href="#3-gradient-descent" class="headerlink" title="3.gradient descent"></a>3.gradient descent</h4><p>pass</p>
<h4 id="Hw1"><a href="#Hw1" class="headerlink" title="Hw1"></a>Hw1</h4><p>pass</p>
<h3 id="W2"><a href="#W2" class="headerlink" title="W2"></a>W2</h3><h4 id="1-bias-and-variance"><a href="#1-bias-and-variance" class="headerlink" title="1.bias and variance"></a>1.bias and variance</h4><p>pass</p>
<p>####2.classification: probabilistic generative model</p>
<p>pass</p>
<h4 id="3-logistic-regression"><a href="#3-logistic-regression" class="headerlink" title="3.logistic regression"></a>3.logistic regression</h4><p>pass</p>
<h3 id="W3"><a href="#W3" class="headerlink" title="W3"></a>W3</h3><p>####1.deep learning</p>
<p>pass</p>
<p>####2.backpropagation</p>
<p>pass</p>
<h4 id="3-Anomaly-detection-异常检测"><a href="#3-Anomaly-detection-异常检测" class="headerlink" title="3.Anomaly detection(异常检测)"></a>3.Anomaly detection(异常检测)</h4><p>pass</p>
<h3 id="W4"><a href="#W4" class="headerlink" title="W4"></a>W4</h3><p>####1.CNN</p>
<p>pass</p>
<h4 id="2-keras"><a href="#2-keras" class="headerlink" title="2.keras"></a>2.keras</h4><p>keras一些文档:<a href="http://keras.io/" target="_blank" rel="noopener">Documentation</a> ,<a href="https://github.com/fchollet/keras/tree/master/examples" target="_blank" rel="noopener">Example</a>,<a href="https://faroit.github.io/keras-docs/2.0.2/getting-started/faq/#how-can-i-save-a-keras-model" target="_blank" rel="noopener">save and load model</a>;</p>
<p>keras安装（参考<a href="https://morvanzhou.github.io/tutorials/machine-learning/keras/1-2-install/" target="_blank" rel="noopener">莫烦教程</a>）</p>
<p>1.安装tensorflow；</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span>1.安装virtualenv</span><br><span class="line">sudo pip install --upgrade virtualenv</span><br><span class="line"><span class="meta">#</span>2.建立一个全新的 virtualenv 环境. 将环境建在 ~/tensorflow 目录下</span><br><span class="line">virtualenv --system-site-packages ~/tensorflow</span><br><span class="line">cd ~/tensorflow</span><br><span class="line"><span class="meta">#</span>3.激活virtualenv</span><br><span class="line">source bin/activate</span><br><span class="line"><span class="meta">#</span>4.在virtualenv下安装tensorflow</span><br><span class="line">pip install --upgrade tensorflow #1.7</span><br><span class="line">pip install --upgrade https://storage.googleapis.com/tensorflow/mac/cpu/tensorflow-1.6.0-py3-none-any.whl #1.6</span><br><span class="line">/Users/zhonghaolin/tensorflow/lib/python3.6/site-packages/tensorflow</span><br></pre></td></tr></table></figure>
<p>2.安装keras，keras依赖tensorflow/Theano作为后端引擎；</p>
<p><code>pip install keras</code></p>
<p>3.使用keras图像识别Mnist</p>
<p><code>mnist.load_data()</code></p>
<p>数据集在<code>~/.keras/datasets/mnist.npz</code></p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">import</span> numpy <span class="keyword">as</span> np</span><br><span class="line"><span class="keyword">from</span> keras.models <span class="keyword">import</span> Sequential</span><br><span class="line"><span class="keyword">from</span> keras.layers.core <span class="keyword">import</span> Dense, Dropout, Activation</span><br><span class="line"><span class="keyword">from</span> keras.layers <span class="keyword">import</span> Conv2D, MaxPooling2D, Flatten</span><br><span class="line"><span class="keyword">from</span> keras.optimizers <span class="keyword">import</span> SGD, Adam</span><br><span class="line"><span class="keyword">from</span> keras.utils <span class="keyword">import</span> np_utils</span><br><span class="line"><span class="keyword">from</span> keras.datasets <span class="keyword">import</span> mnist</span><br><span class="line"><span class="comment">#categorical_crossentropy</span></span><br><span class="line"></span><br><span class="line"><span class="function"><span class="keyword">def</span> <span class="title">load_data</span><span class="params">()</span>:</span></span><br><span class="line">    (x_train, y_train), (x_test, y_test) = mnist.load_data()</span><br><span class="line">    number = <span class="number">10000</span></span><br><span class="line">    x_train = x_train[<span class="number">0</span>:number]</span><br><span class="line">    y_train = y_train[<span class="number">0</span>:number]</span><br><span class="line">    x_train = x_train.reshape(number, <span class="number">28</span>*<span class="number">28</span>)</span><br><span class="line">    x_test = x_test.reshape(number, <span class="number">28</span>*<span class="number">28</span>)</span><br><span class="line">    x_train = x_train.astype(<span class="string">'float32'</span>)</span><br><span class="line">    x_test = x_test.astype(<span class="string">'float32'</span>)</span><br><span class="line">    <span class="comment"># convert class vectors to binary class </span></span><br><span class="line">    y_train = np_utils.to_categorical(y_train, <span class="number">10</span>)</span><br><span class="line">    y_test = np_utils.to_categorical(y_test, <span class="number">10</span>)</span><br><span class="line">    x_train = x_train</span><br><span class="line">    y_train = y_train</span><br><span class="line">    <span class="comment">#</span></span><br><span class="line">    x_train = x_train/<span class="number">255</span></span><br><span class="line">    x_test = x_test/<span class="number">255</span></span><br><span class="line">    <span class="keyword">return</span> (x_train, y_train), (x_test, y_test)</span><br><span class="line"></span><br><span class="line">(x_train, y_train), (x_test, y_test) = load_data()</span><br><span class="line">model=Sequential()</span><br><span class="line">model.add(Dense(input_dim=<span class="number">28</span>*<span class="number">28</span>, units=<span class="number">600</span>, activation=<span class="string">'sigmoid'</span>))</span><br><span class="line">model.add(Dense(units=<span class="number">600</span>, activation=<span class="string">'sigmoid'</span>))</span><br><span class="line">model.add(Dense(units=<span class="number">600</span>, activation=<span class="string">'sigmoid'</span>))</span><br><span class="line">model.add(Dense(units=<span class="number">10</span>, activation=<span class="string">'softmax'</span>))</span><br><span class="line">model.compile(loss=<span class="string">'mse'</span>, optimizer=SGD(lr=<span class="number">0.1</span>), metrics=[<span class="string">'accuracy'</span>])</span><br><span class="line">model.fit(x_train, y_train, batch_size=<span class="number">10</span>, epochs=<span class="number">20</span>)</span><br><span class="line">result = model.evaluate(x_test, y_test)</span><br><span class="line">print(<span class="string">"\nTest Acc:"</span>, result[<span class="number">1</span>])</span><br></pre></td></tr></table></figure>
<h4 id="HW3"><a href="#HW3" class="headerlink" title="HW3"></a>HW3</h4><p>作业相关：<a href="https://docs.google.com/presentation/d/1_UOEquMW4URIuTtrKGqwSPP4qtKIb7osT9ysTQ3Flmo/edit#slide=id.g28fb825f54_0_939" target="_blank" rel="noopener">ml-2019spring-hw3</a>,<a href="https://hackmd.io/hlJAABuoRJa0tzpD8UUHvw" target="_blank" rel="noopener">示例</a> , <a href="https://www.kaggle.com/c/ml2019spring-hw3/overview" target="_blank" rel="noopener">kaggle</a> ;</p>
<p>keras搭建CNN；（参考<a href="https://blog.csdn.net/u012458963/article/details/72230729）" target="_blank" rel="noopener">https://blog.csdn.net/u012458963/article/details/72230729）</a></p>
<h4 id="HW4"><a href="#HW4" class="headerlink" title="HW4"></a>HW4</h4><p>作业相关：<a href="https://ntumlta2019.github.io/ml-web-hw4/" target="_blank" rel="noopener">ml-2019spring-hw4</a> ,<a href="https://hackmd.io/435VaJPrQtCJzAdodxXbxA" target="_blank" rel="noopener">示例</a>;</p>
<p>1.keras实现saliency map；(参考<a href="https://fairyonice.github.io/Saliency-Map-with-keras-vis.html" target="_blank" rel="noopener">https://fairyonice.github.io/Saliency-Map-with-keras-vis.html</a> ；<a href="https://mathpretty.com/10683.html；https://blog.csdn.net/qq576826394/article/details/90298003" target="_blank" rel="noopener">https://mathpretty.com/10683.html；https://blog.csdn.net/qq576826394/article/details/90298003</a>)</p>
<p>安装pytorch(使用conda安装在主环境上了)安装1.2版本的，更高版本貌似<a href="https://heary.cn/posts/PyTorch%E6%8A%A5CUDA-error-no-kernel-image-is-available-for-execution-on-the-device%E9%97%AE%E9%A2%98%E8%A7%A3%E5%86%B3/" target="_blank" rel="noopener">不兼容Tesla K40m显卡</a>；</p>
<p><code>conda install pytorch==1.2.0 torchvision==0.4.0 cudatoolkit=9.2 -c pytorch</code></p>
<p><code>pip install torch==1.2.0+cu92 torchvision==0.4.0+cu92 -f  https://download.pytorch.org/whl/torch_stable.html</code></p>
<p><a href="https://pytorch.org/get-started/locally/" target="_blank" rel="noopener">Pytorch官网</a>选择系统，输入命令如下</p>
<p><img src="/Users/zhonghaolin/Hexo/source/_posts/image/install_pytorch.png" alt=""></p>
<p>2.<a href="https://github.com/marcotcr/lime" target="_blank" rel="noopener">Lime</a></p>
<h3 id="W5"><a href="#W5" class="headerlink" title="W5"></a>W5</h3><p>pass</p>
<h3 id="W6"><a href="#W6" class="headerlink" title="W6"></a>W6</h3><p>Pass</p>

          
        
      
    </div>
    
    
    

    

    

    

    <footer class="post-footer">
      

      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </div>
  
  
  
  </article>


    
      

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://yoursite.com/2020/02/28/model_organism_proteomes/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="HaolinZhong">
      <meta itemprop="description" content="">
      <meta itemprop="image" content="https://timgsa.baidu.com/timg?image&quality=80&size=b9999_10000&sec=1534345547319&di=b9e880681b19e0cefb3132f38ccb0281&imgtype=0&src=http%3A%2F%2Fimg.duoziwang.com%2F2017%2F09%2F1607450267817.jpg">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="find my way">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
                
                <a class="post-title-link" href="/2020/02/28/model_organism_proteomes/" itemprop="url">human_proteomes</a></h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2020-02-28T16:03:54+08:00">
                2020-02-28
              </time>
            

            

            
          </span>

          

          
            
          

          
          

          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p>[TOC]</p>
<p>###1.模式生物</p>
<table>
<thead>
<tr>
<th>Kingdom</th>
<th>Model Organism</th>
<th>Common name</th>
<th>Informal classsification</th>
<th>Usage(examples)</th>
</tr>
</thead>
<tbody>
<tr>
<td>Virus</td>
<td><a href="https://en.wikipedia.org/wiki/Phi_X_174" target="_blank" rel="noopener">Phi X 174</a></td>
<td></td>
<td>Virus</td>
<td>Evolution</td>
</tr>
<tr>
<td>Prokaryote</td>
<td><em><a href="https://en.wikipedia.org/wiki/Escherichia_coli" target="_blank" rel="noopener">Escherichia coli</a></em></td>
<td></td>
<td>Bacteria</td>
<td>bacterial genetics, metabolism</td>
</tr>
<tr>
<td>Eukaryote, unicellular</td>
<td><em><a href="https://en.wikipedia.org/wiki/Dictyostelium_discoideum" target="_blank" rel="noopener">Dictyostelium discoideum</a></em></td>
<td></td>
<td><a href="https://en.wikipedia.org/wiki/Amoeba" target="_blank" rel="noopener">Amoeba</a> (阿米巴)</td>
<td></td>
</tr>
<tr>
<td></td>
<td><em><a href="https://en.wikipedia.org/wiki/Saccharomyces_cerevisiae" target="_blank" rel="noopener">Saccharomyces cerevisiae</a></em></td>
<td></td>
<td>Yeast</td>
<td>cell division, organelles, etc.</td>
</tr>
<tr>
<td></td>
<td><em><a href="https://en.wikipedia.org/wiki/Schizosaccharomyces_pombe" target="_blank" rel="noopener">Schizosaccharomyces pombe</a></em></td>
<td></td>
<td>Yeast</td>
<td>cell cycle, cytokinesis, chromosome biology, telomeres, DNA metabolism, cytoskeleton organization, industrial applications</td>
</tr>
<tr>
<td></td>
<td><em><a href="https://en.wikipedia.org/wiki/Chlamydomonas_reinhardtii" target="_blank" rel="noopener">Chlamydomonas reinhardtii</a></em></td>
<td></td>
<td><a href="https://en.wikipedia.org/wiki/Algae" target="_blank" rel="noopener">Algae</a> (海藻)</td>
<td></td>
</tr>
<tr>
<td></td>
<td><em><a href="https://en.wikipedia.org/wiki/Tetrahymena_thermophila" target="_blank" rel="noopener">Tetrahymena thermophila</a></em></td>
<td></td>
<td><a href="https://en.wikipedia.org/wiki/Ciliate" target="_blank" rel="noopener">Ciliate</a> （纤毛虫）</td>
<td></td>
</tr>
<tr>
<td></td>
<td><em><a href="https://en.wikipedia.org/wiki/Emiliania_huxleyi" target="_blank" rel="noopener">Emiliania huxleyi</a></em></td>
<td></td>
<td><a href="https://en.wikipedia.org/wiki/Plankton" target="_blank" rel="noopener">Plankton</a> （浮游生物）</td>
<td></td>
</tr>
<tr>
<td>Eukaryote, multicellular</td>
<td><em><a href="https://en.wikipedia.org/wiki/Caenorhabditis_elegans" target="_blank" rel="noopener">Caenorhabditis elegans</a></em></td>
<td></td>
<td><a href="https://en.wikipedia.org/wiki/Worm" target="_blank" rel="noopener">Worm</a></td>
<td>differentiation, development</td>
</tr>
<tr>
<td></td>
<td><em><a href="https://en.wikipedia.org/wiki/Drosophila_melanogaster" target="_blank" rel="noopener">Drosophila melanogaster</a></em></td>
<td>Fruit fly</td>
<td><a href="https://en.wikipedia.org/wiki/Insect" target="_blank" rel="noopener">Insect</a></td>
<td>developmental biology, human brain degenerative disease</td>
</tr>
<tr>
<td></td>
<td><em><a href="https://en.wikipedia.org/wiki/Callosobruchus_maculatus" target="_blank" rel="noopener">Callosobruchus maculatus</a></em></td>
<td>Cowpea Weevil</td>
<td><a href="https://en.wikipedia.org/wiki/Insect" target="_blank" rel="noopener">Insect</a></td>
<td>developmental biology</td>
</tr>
<tr>
<td></td>
<td><em><a href="https://en.wikipedia.org/wiki/Arabidopsis_thaliana" target="_blank" rel="noopener">Arabidopsis thaliana</a></em></td>
<td>Thale cress</td>
<td><a href="https://en.wikipedia.org/wiki/Flowering_plant" target="_blank" rel="noopener">Flowering plant</a></td>
<td></td>
</tr>
<tr>
<td></td>
<td><em><a href="https://en.wikipedia.org/wiki/Physcomitrella_patens" target="_blank" rel="noopener">Physcomitrella patens</a></em></td>
<td>Spreading earthmoss</td>
<td><a href="https://en.wikipedia.org/wiki/Moss" target="_blank" rel="noopener">Moss</a></td>
<td></td>
</tr>
<tr>
<td>Vertebrate</td>
<td><em><a href="https://en.wikipedia.org/wiki/Danio_rerio" target="_blank" rel="noopener">Danio rerio</a></em></td>
<td>Zebrafish</td>
<td><a href="https://en.wikipedia.org/wiki/Fish" target="_blank" rel="noopener">Fish</a></td>
<td>embryonic development</td>
</tr>
<tr>
<td></td>
<td><em><a href="https://en.wikipedia.org/wiki/Mummichog" target="_blank" rel="noopener">Fundulus heteroclitus</a></em></td>
<td>Mummichog</td>
<td>Fish</td>
<td></td>
</tr>
<tr>
<td></td>
<td><em><a href="https://en.wikipedia.org/wiki/Nothobranchius_furzeri" target="_blank" rel="noopener">Nothobranchius furzeri</a></em></td>
<td>Turquoise killifish</td>
<td>Fish</td>
<td>aging, disease, evolution</td>
</tr>
<tr>
<td></td>
<td><em><a href="https://en.wikipedia.org/wiki/Japanese_rice_fish" target="_blank" rel="noopener">Oryzias latipes</a></em></td>
<td>Japanese rice fish</td>
<td>Fish</td>
<td></td>
</tr>
<tr>
<td></td>
<td><em><a href="https://en.wikipedia.org/wiki/Anolis_carolinensis" target="_blank" rel="noopener">Anolis carolinensis</a></em></td>
<td>Carolina anole</td>
<td><a href="https://en.wikipedia.org/wiki/Reptile" target="_blank" rel="noopener">Reptile</a></td>
<td>reptile biology, evolution</td>
</tr>
<tr>
<td></td>
<td><em><a href="https://en.wikipedia.org/wiki/Mus_musculus" target="_blank" rel="noopener">Mus musculus</a></em></td>
<td>House mouse</td>
<td><a href="https://en.wikipedia.org/wiki/Mammal" target="_blank" rel="noopener">Mammal</a></td>
<td>disease model for humans</td>
</tr>
<tr>
<td></td>
<td><em><a href="https://en.wikipedia.org/wiki/Gallus_gallus" target="_blank" rel="noopener">Gallus gallus</a></em></td>
<td>Red junglefowl</td>
<td><a href="https://en.wikipedia.org/wiki/Bird" target="_blank" rel="noopener">Bird</a></td>
<td>embryological development and organogenesis</td>
</tr>
<tr>
<td></td>
<td><em><a href="https://en.wikipedia.org/wiki/Xenopus_laevis" target="_blank" rel="noopener">Xenopus laevis</a></em> (Note: and <em>X. tropicalis</em>)</td>
<td>African clawed frog</td>
<td><a href="https://en.wikipedia.org/wiki/Amphibian" target="_blank" rel="noopener">Amphibian</a></td>
<td>embryonic development·</td>
</tr>
</tbody>
</table>
<p>reference：<a href="https://en.wikipedia.org/wiki/Model_organism" target="_blank" rel="noopener">https://en.wikipedia.org/wiki/Model_organism</a></p>
<p>其中常用的有<strong><em>Saccharomyces cerevisiae</em></strong>，<strong><em>Escherichia coli</em></strong> ，<strong><em>Caenorhabditis elegans</em></strong>，<strong><em>Drosophila melanogaster</em></strong>，<strong><em>Arabidopsis thaliana</em></strong>，<strong><em>Danio rerio</em></strong>，<strong><em>Mus musculus</em></strong>，<strong><em>Xenopus laevis</em></strong>;</p>
<h3 id="2-homo-sapiens"><a href="#2-homo-sapiens" class="headerlink" title="2.homo sapiens"></a>2.homo sapiens</h3><p>1.human proteomes download</p>
<p><a href="https://www.uniprot.org/proteomes/?query=homo%20sapiens&amp;fil=reference%3Ayes&amp;sort=score" target="_blank" rel="noopener">Uniprot Proteomes</a> 搜索条件：homo sapiens AND reference:yes</p>
<p>搜索结果如下：</p>
<p><img src="/Users/zhonghaolin/Library/Application Support/typora-user-images/image-20200228160724346.png" alt="image-20200228160724346"></p>
<p>下载第一个protein count有74811的UP000005640, <strong>Organism ID</strong>:9606</p>
<p>点击Protein Count：74811，下载fasta序列；</p>
<p>2.Uniprot注释下载</p>
<p>限定物种和有结构域注释进行搜索，text格式下载所有</p>
<p><img src="/Users/zhonghaolin/Library/Application Support/typora-user-images/image-20200306105813268.png" alt="image-20200306105813268"></p>
<p>3.从有注释的序列中提取出人类蛋白质组的序列</p>
<p>格式如下：</p>
<table>
<thead>
<tr>
<th>entry</th>
<th>domain</th>
<th>evidence</th>
<th>length</th>
</tr>
</thead>
<tbody>
<tr>
<td>Q8N8B7</td>
<td>[‘5-82’, ‘173-289’]</td>
<td>[‘0000255’, ‘0000255’]</td>
<td>351</td>
</tr>
<tr>
<td>O75494</td>
<td>[‘10-88’]</td>
<td>[‘0000255’]</td>
<td>262</td>
</tr>
</tbody>
</table>
<p>ECO(evidence conclusion ontology)说明如下：</p>
<table>
<thead>
<tr>
<th>ECO</th>
<th>description</th>
<th>Num</th>
</tr>
</thead>
<tbody>
<tr>
<td>0000259</td>
<td>Sequence model evidence(automatic assertions for information which has been generated by the UniProtKB <a href="https://www.uniprot.org/help/automatic_annotation" target="_blank" rel="noopener">automatic annotation</a> system)</td>
<td>41502</td>
</tr>
<tr>
<td>0000255</td>
<td>Sequence model evidence(manual assertions for information which has been generated by the UniProtKB <a href="https://www.uniprot.org/help/automatic_annotation" target="_blank" rel="noopener">automatic annotation</a> system)</td>
<td>16547</td>
</tr>
<tr>
<td>None</td>
<td>没有evidence来源说明</td>
<td>3775</td>
</tr>
<tr>
<td>0000305</td>
<td>Curator inference evidence(manually curated information which has been inferred by a curator based  on his/her scientific knowledge or on the scientific content of an  article.)</td>
<td>22</td>
</tr>
<tr>
<td>0000250</td>
<td>Sequence similarity evidence(for manually curated information which has been propagated from a related experimentally characterized protein.)</td>
<td>10</td>
</tr>
<tr>
<td>0000269</td>
<td>Experimental evidence</td>
<td>3</td>
</tr>
<tr>
<td>0000244</td>
<td>Combinatorial evidence(in manual assertions for information inferred from a combination of experimental and computational evidence.)</td>
<td>2</td>
</tr>
</tbody>
</table>
<p>Uniprot ECO(evidence conclusion ontology)解释说明：<a href="https://www.uniprot.org/help/evidences" target="_blank" rel="noopener">https://www.uniprot.org/help/evidences</a></p>
<p>4.统计结果</p>
<table>
<thead>
<tr>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>73928</td>
<td>40305</td>
<td>14912</td>
<td>3775</td>
</tr>
</tbody>
</table>
<p>筛选长度在[80, 2000]的序列</p>
<table>
<thead>
<tr>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>73928</td>
<td>30082</td>
<td>14250</td>
<td>3196</td>
</tr>
</tbody>
</table>
<p>2.序列处理</p>
<p>​    2.1split</p>
<p>把下载下来的蛋白质组按照id拆分成一条一条的fasta序列</p>
<p>​    2.2filter</p>
<p>统计序列长度如下：</p>
<table>
<thead>
<tr>
<th>num(l&lt;80)</th>
<th>num(80=&lt;l&lt;2000)</th>
<th>num(l&gt;2000)</th>
</tr>
</thead>
<tbody>
<tr>
<td>11951</td>
<td>61187</td>
<td>789</td>
</tr>
</tbody>
</table>
<p>保留长度在80-2000的序列61187条</p>
<p>​    2.3处理含有X字符的序列</p>
<p>I-Tasser不能处理含有X的蛋白序列，会报错，删除序列中的X字符</p>
<p>​    2.4任务提交</p>
<p>从序列列表里每次读区1000条序列，然后分配好提交到节点进行计算</p>
<p>3.结构域预测</p>
<p>​    3.1使用I-TASSER得到init file</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span>I-TASSER版本5.1</span><br><span class="line"><span class="meta">#</span>I-TASSER dir：/home/haolinz/bio/bio_software/I-TASSER5.1/</span><br><span class="line"><span class="meta">#</span>因为只需要得到初始文件，所以只用跑到3.1，把后面的部分注释掉；</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span>output dir : seqname/*</span><br><span class="line">/home/haolinz/bio/bio_software/I-TASSER5.1/I-TASSERmod/runI-TASSER_mkinit_zhl.pl -pkgdir /home/haolinz/bio/bio_software/I-TASSER5.1 -libdir /nfs/ITLIB/ -LBS true -EC true -GO true -seqname $chainId -datadir /tmp/$user/$jobId/$chainId -light true -outdir /tmp/$user/$jobId/$chainId</span><br></pre></td></tr></table></figure>
<p>​    3.2用inint file跑ThreaDom</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">./ThreaDom.pl -seqname B4DNK4 -datadir /home/haolinz/workspace/I-Tasser_human/output/B4DNK4 -tlibdir /home/haolinz/bio/bio_software/threadom -libdir /nfs/ITLIB/ -cutoff 0.6 0.76 0.8</span><br></pre></td></tr></table></figure>
<p>###3.mus musculus(mouse)</p>
<p>1.小鼠蛋白质组fasta下载</p>
<p><img src="/Users/zhonghaolin/Library/Application Support/typora-user-images/image-20200304160520445.png" alt="image-20200304160520445"></p>
<p>下载<a href="https://www.uniprot.org/proteomes/UP000000589" target="_blank" rel="noopener">UP000000589</a> , Organism ID: 10090;</p>
<p>2.Uniprot注释下载</p>
<p><img src="/Users/zhonghaolin/Library/Application Support/typora-user-images/image-20200305172739717.png" alt="image-20200305172739717"></p>
<p>3.统计结果</p>
<table>
<thead>
<tr>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>55421</td>
<td>27985</td>
<td>12828</td>
<td>3419</td>
</tr>
</tbody>
</table>
<p>筛选长度在[80, 2000]的序列</p>
<table>
<thead>
<tr>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>55421</td>
<td>22423</td>
<td>12152</td>
<td>2854</td>
</tr>
</tbody>
</table>
<h3 id="4-Saccharomyces-cerevisiae"><a href="#4-Saccharomyces-cerevisiae" class="headerlink" title="4.Saccharomyces cerevisiae"></a>4.Saccharomyces cerevisiae</h3><p>下载<a href="https://www.uniprot.org/proteomes/UP000002311" target="_blank" rel="noopener">UP000002311</a>,Organism ID：559292</p>
<p>6049条</p>
<table>
<thead>
<tr>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>6049</td>
<td>4503</td>
<td>795</td>
<td>425</td>
</tr>
</tbody>
</table>
<p>筛选长度在[80, 2000]的序列</p>
<table>
<thead>
<tr>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>6049</td>
<td>4254</td>
<td>778</td>
<td>416</td>
</tr>
</tbody>
</table>
<h3 id="5-Escherichia-coli"><a href="#5-Escherichia-coli" class="headerlink" title="5.Escherichia coli"></a>5.Escherichia coli</h3><p>下载<a href="https://www.uniprot.org/proteomes/UP000000625" target="_blank" rel="noopener">UP000000625</a>,Organism ID：83333</p>
<p>4391条</p>
<table>
<thead>
<tr>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>4391</td>
<td>3386</td>
<td>372</td>
<td>188</td>
</tr>
</tbody>
</table>
<p>筛选长度在[80, 2000]的序列</p>
<table>
<thead>
<tr>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>4391</td>
<td>3046</td>
<td>372</td>
<td>179</td>
</tr>
</tbody>
</table>
<h3 id="6-Arabidopsis-thaliana"><a href="#6-Arabidopsis-thaliana" class="headerlink" title="6.Arabidopsis thaliana"></a>6.Arabidopsis thaliana</h3><p>下载<a href="https://www.uniprot.org/proteomes/UP000006548" target="_blank" rel="noopener">UP000006548</a>,Organism ID: 3702</p>
<p>39359条</p>
<table>
<thead>
<tr>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>39359</td>
<td>20702</td>
<td>10526</td>
<td>1992</td>
</tr>
</tbody>
</table>
<p>筛选长度在[80, 2000]的序列</p>
<table>
<thead>
<tr>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>39359</td>
<td>19285</td>
<td>10403</td>
<td>1980</td>
</tr>
</tbody>
</table>
<p>###7.Caenorhabditis elegans</p>
<p>下载<a href="https://www.uniprot.org/proteomes/UP000001940" target="_blank" rel="noopener">UP000001940</a> ,Organism ID:6239</p>
<p>26955/26850</p>
<table>
<thead>
<tr>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>26955</td>
<td>14681</td>
<td>6865</td>
<td>444</td>
</tr>
</tbody>
</table>
<p>筛选长度在[80, 2000]的序列</p>
<table>
<thead>
<tr>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>26955</td>
<td>13266</td>
<td>6570</td>
<td>326</td>
</tr>
</tbody>
</table>
<h3 id="8-Drosophila-melanogaster"><a href="#8-Drosophila-melanogaster" class="headerlink" title="8.Drosophila melanogaster"></a>8.Drosophila melanogaster</h3><p>下载<a href="https://www.uniprot.org/proteomes/UP000000803" target="_blank" rel="noopener">UP000000803</a> ,Organism ID:7227</p>
<p>21973</p>
<table>
<thead>
<tr>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>21973</td>
<td>9623</td>
<td>7950</td>
<td>417</td>
</tr>
</tbody>
</table>
<p>筛选长度在[80, 2000]的序列</p>
<table>
<thead>
<tr>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>21973</td>
<td>9009</td>
<td>7232</td>
<td>332</td>
</tr>
</tbody>
</table>
<p>###9.Danio rerio(zebra fish)</p>
<p>下载<a href="https://www.uniprot.org/proteomes/UP000000437" target="_blank" rel="noopener">UP000000437</a> ,Organism ID:7955</p>
<p>46847/46848</p>
<table>
<thead>
<tr>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>46847</td>
<td>15510</td>
<td>17136</td>
<td>438</td>
</tr>
</tbody>
</table>
<p>筛选长度在[80, 2000]的序列</p>
<table>
<thead>
<tr>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>46487</td>
<td>14066</td>
<td>16433</td>
<td>396</td>
</tr>
</tbody>
</table>
<h3 id="10-Xenopus-laevis"><a href="#10-Xenopus-laevis" class="headerlink" title="10.Xenopus laevis"></a>10.Xenopus laevis</h3><p>下载<a href="https://www.uniprot.org/proteomes/UP000186698" target="_blank" rel="noopener">UP000186698</a> ,Organism ID: 8355</p>
<p>44571</p>
<table>
<thead>
<tr>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>44571</td>
<td>19916</td>
<td>13751</td>
<td>0</td>
</tr>
</tbody>
</table>
<p>筛选长度在[80, 2000]的序列</p>
<table>
<thead>
<tr>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>44571</td>
<td>16521</td>
<td>13201</td>
<td>0</td>
</tr>
</tbody>
</table>
<p>统计如下</p>
<p><img src="/Users/zhonghaolin/Library/Application Support/typora-user-images/image-20200309233852479.png" alt="image-20200309233852479"></p>
<p>模式生物注释统计汇总</p>
<table>
<thead>
<tr>
<th>物种</th>
<th>总序列数</th>
<th>无注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
</tr>
</thead>
<tbody>
<tr>
<td>homo sapiens</td>
<td>73928</td>
<td>40305</td>
<td>14912</td>
<td>3775</td>
</tr>
<tr>
<td>mus musculus</td>
<td>55421</td>
<td>27985</td>
<td>12828</td>
<td>3419</td>
</tr>
<tr>
<td>Saccharomyces cerevisiae</td>
<td>6049</td>
<td>4503</td>
<td>795</td>
<td>425</td>
</tr>
<tr>
<td>Escherichia coli</td>
<td>4391</td>
<td>3386</td>
<td>372</td>
<td>188</td>
</tr>
<tr>
<td>Arabidopsis thaliana</td>
<td>39359</td>
<td>20702</td>
<td>10526</td>
<td>1992</td>
</tr>
<tr>
<td>Caenorhabditis elegans</td>
<td>26955</td>
<td>14681</td>
<td>6865</td>
<td>444</td>
</tr>
<tr>
<td>Drosophila melanogaster</td>
<td>21973</td>
<td>9623</td>
<td>7950</td>
<td>417</td>
</tr>
<tr>
<td>Danio rerio</td>
<td>46847</td>
<td>15510</td>
<td>17136</td>
<td>438</td>
</tr>
<tr>
<td>Xenopus laevis</td>
<td>44571</td>
<td>19916</td>
<td>13751</td>
<td>0</td>
</tr>
</tbody>
</table>
<table>
<thead>
<tr>
<th>物种</th>
<th>总序列数</th>
<th>无结构域划分注释</th>
<th>有结构域划分注释</th>
<th>注释的结构域长度小于总长50%</th>
<th>无来源注释</th>
<th>无注释/总序列</th>
<th>Reviewed</th>
<th>有功能注释的</th>
</tr>
</thead>
<tbody>
<tr>
<td>homo sapiens</td>
<td>73928</td>
<td>40305</td>
<td>33623</td>
<td>14912</td>
<td>3775</td>
<td>0.54519262</td>
<td>20350</td>
<td>70088</td>
</tr>
<tr>
<td>mus musculus</td>
<td>55421</td>
<td>27985</td>
<td>27436</td>
<td>12828</td>
<td>3419</td>
<td>0.504952996</td>
<td>17034</td>
<td>51216</td>
</tr>
<tr>
<td>Saccharomyces cerevisiae</td>
<td>6049</td>
<td>4503</td>
<td>1546</td>
<td>795</td>
<td>425</td>
<td>0.744420565</td>
<td>6049</td>
<td>6049</td>
</tr>
<tr>
<td>Escherichia coli</td>
<td>4391</td>
<td>3386</td>
<td>1005</td>
<td>372</td>
<td>188</td>
<td>0.771122751</td>
<td>4389</td>
<td>4390</td>
</tr>
<tr>
<td>Arabidopsis thaliana</td>
<td>39359</td>
<td>20702</td>
<td>18657</td>
<td>10526</td>
<td>1992</td>
<td>0.52597881</td>
<td>15877</td>
<td>35692</td>
</tr>
<tr>
<td>Caenorhabditis elegans</td>
<td>26955</td>
<td>14681</td>
<td>12274</td>
<td>6865</td>
<td>444</td>
<td>0.544648488</td>
<td>4128</td>
<td>23969</td>
</tr>
<tr>
<td>Drosophila melanogaster</td>
<td>21973</td>
<td>9623</td>
<td>12350</td>
<td>7950</td>
<td>417</td>
<td>0.437946571</td>
<td>3576</td>
<td>20644</td>
</tr>
<tr>
<td>Danio rerio</td>
<td>46847</td>
<td>15510</td>
<td>31337</td>
<td>17136</td>
<td>438</td>
<td>0.331077764</td>
<td>3152</td>
<td>44050</td>
</tr>
<tr>
<td>Xenopus laevis</td>
<td>44571</td>
<td>19916</td>
<td>24655</td>
<td>13751</td>
<td>0</td>
<td>0.44683763</td>
<td>39</td>
<td>36905</td>
</tr>
</tbody>
</table>
<p>ecoli    ✅</p>
<p>nohup python Saccharomyces\ cerevisiae.py &gt; Saccharomyces\ cerevisiae.log 2&gt;&amp;1 &amp;    ✅</p>
<p>nohup python Drosophila\ melanogaster.py &gt; Drosophila\ melanogaster.log 2&gt;&amp;1 &amp;   ✅</p>
<p>nohup python homo\ sapiens.py &gt;&gt; homo\ sapiens_anno.log 2&gt;&amp;1 &amp;    run  35101 ✅</p>
<p>nohup python mus\ musculus.py &gt;&gt; mus\ musculus_anno.log 2&gt;&amp;1 &amp;  run   41191  46759  ✅</p>
<p>nohup python Arabidopsis\ thaliana.py &gt;&gt; Arabidopsis\ thaliana_anno.log 2&gt;&amp;1 &amp; run ✅</p>
<p>nohup python Caenorhabditis\ elegans.py &gt;&gt; Caenorhabditis\ elegans_anno.log 2&gt;&amp;1 &amp;  run 14942 ✅</p>
<p>UniProt <a href="https://www.uniprot.org/help/protein_existence" target="_blank" rel="noopener">protein existence</a> statistic</p>
<p>‘PE   5: Uncertain;\n’, ‘PE   4: Predicted;\n’, ‘PE   2: Evidence at transcript level;\n’, ‘PE   3: Inferred from homology;\n’, ‘PE   1: Evidence at protein level;’</p>
<p>1: Evidence at protein level，2: Evidence at transcript level，3: Inferred from homology，4: Predicted，5: Uncertain</p>
<p>20200519 人类、小鼠、拟南芥、线虫、斑马鱼蛋白质组序列数量发生变化；</p>
<table>
<thead>
<tr>
<th>物种</th>
<th>总序列数</th>
<th>gene count</th>
<th>Swiss-Prot</th>
<th>TrEMBL</th>
<th>1</th>
<th>2</th>
<th>3</th>
<th>4</th>
<th>5</th>
</tr>
</thead>
<tbody>
<tr>
<td>homo sapiens</td>
<td>74823</td>
<td>20596</td>
<td>20350</td>
<td></td>
<td>16265</td>
<td>2594</td>
<td>752</td>
<td>147</td>
<td>592</td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td>54473</td>
<td>43709</td>
<td>315</td>
<td>743</td>
<td>9706</td>
<td>0</td>
</tr>
<tr>
<td>mus musculus</td>
<td>55398</td>
<td>21982</td>
<td>17034</td>
<td></td>
<td>12670</td>
<td>3910</td>
<td>378</td>
<td>67</td>
<td>9</td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td>38364</td>
<td>23783</td>
<td>1903</td>
<td>2235</td>
<td>10443</td>
<td>0</td>
</tr>
<tr>
<td>Saccharomyces cerevisiae</td>
<td>6049</td>
<td>6049</td>
<td>6049</td>
<td>0</td>
<td>5259</td>
<td>130</td>
<td>261</td>
<td>372</td>
<td>27</td>
</tr>
<tr>
<td>Escherichia coli</td>
<td>4391</td>
<td>4391</td>
<td>4389</td>
<td></td>
<td>3036</td>
<td>161</td>
<td>656</td>
<td>437</td>
<td>99</td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td>2</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>0</td>
</tr>
<tr>
<td>Arabidopsis thaliana</td>
<td>39364</td>
<td>27463</td>
<td>15877</td>
<td></td>
<td>6936</td>
<td>6731</td>
<td>1606</td>
<td>486</td>
<td>118</td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td>23487</td>
<td>1748</td>
<td>5699</td>
<td>2645</td>
<td>13386</td>
<td>0</td>
</tr>
<tr>
<td>Caenorhabditis elegans</td>
<td>26850</td>
<td>19867</td>
<td>4128</td>
<td></td>
<td>1670</td>
<td>659</td>
<td>1358</td>
<td>434</td>
<td>7</td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td>22722</td>
<td>10448</td>
<td>184</td>
<td>1253</td>
<td>10837</td>
<td>0</td>
</tr>
<tr>
<td>Drosophila melanogaster</td>
<td>21973</td>
<td>13790</td>
<td>3576</td>
<td></td>
<td>1983</td>
<td>1410</td>
<td>172</td>
<td>11</td>
<td>0</td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td>18397</td>
<td>6301</td>
<td>4947</td>
<td>1357</td>
<td>5792</td>
<td>0</td>
</tr>
<tr>
<td>Danio rerio</td>
<td>46848</td>
<td>25706</td>
<td>3152</td>
<td></td>
<td>417</td>
<td>2375</td>
<td>352</td>
<td>8</td>
<td>0</td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td>43696</td>
<td>12800</td>
<td>2849</td>
<td>5090</td>
<td>22957</td>
<td>0</td>
</tr>
<tr>
<td>Xenopus laevis</td>
<td>44571</td>
<td>43235</td>
<td>39</td>
<td></td>
<td>17</td>
<td>15</td>
<td>7</td>
<td>0</td>
<td>0</td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td>44532</td>
<td>4784</td>
<td>2368</td>
<td>7301</td>
<td>30079</td>
<td>0</td>
</tr>
</tbody>
</table>
<p>Genome3D anno statistic</p>
<table>
<thead>
<tr>
<th>物种</th>
<th>总序列数</th>
<th>有注释</th>
<th>无注释</th>
<th>有注释/总数</th>
</tr>
</thead>
<tbody>
<tr>
<td>homo sapiens</td>
<td>73928</td>
<td>19438</td>
<td>54490</td>
<td><strong><u>26.29%</u></strong></td>
</tr>
<tr>
<td>mus musculus</td>
<td>55421</td>
<td>16285</td>
<td>39127</td>
<td><strong><u>29.39%</u></strong></td>
</tr>
<tr>
<td>Saccharomyces cerevisiae</td>
<td>6049</td>
<td>5775</td>
<td>274</td>
<td>95.47%</td>
</tr>
<tr>
<td>Escherichia coli</td>
<td>4391</td>
<td>4107</td>
<td>284</td>
<td>93.53%</td>
</tr>
<tr>
<td>Arabidopsis thaliana</td>
<td>39359</td>
<td>33840</td>
<td>5519</td>
<td>85.98%</td>
</tr>
<tr>
<td>Caenorhabditis elegans</td>
<td>26955</td>
<td>22427</td>
<td>4528</td>
<td>83.20%</td>
</tr>
<tr>
<td>Drosophila melanogaster</td>
<td>21973</td>
<td>20046</td>
<td>1927</td>
<td>91.23%</td>
</tr>
<tr>
<td>Danio rerio</td>
<td>46847</td>
<td>15510</td>
<td>46847</td>
<td>0</td>
</tr>
<tr>
<td>Xenopus laevis</td>
<td>44571</td>
<td>19916</td>
<td>44571</td>
<td>0</td>
</tr>
</tbody>
</table>
<p>和Genome3D比较就是homo sapiens 和mus musculus还可以注释更加充分，另外补充Danio rerio和Xenopus laevis；Genome 3D多出来的模式生物是疟原虫，金黄葡萄球菌和另一种酵母，另外还注释有猪，牛，小麦，肺结核杆菌；</p>
<table>
<thead>
<tr>
<th>物种</th>
<th>总序列数</th>
<th>gene count</th>
</tr>
</thead>
<tbody>
<tr>
<td>homo sapiens</td>
<td>74823</td>
<td>20596</td>
</tr>
<tr>
<td>mus musculus</td>
<td>55398</td>
<td>21982</td>
</tr>
<tr>
<td>Saccharomyces cerevisiae</td>
<td>6049</td>
<td>6049</td>
</tr>
<tr>
<td>Escherichia coli</td>
<td>4391</td>
<td>4391</td>
</tr>
<tr>
<td>Arabidopsis thaliana</td>
<td>39364</td>
<td>27463</td>
</tr>
<tr>
<td>Caenorhabditis elegans</td>
<td>26850</td>
<td>19867</td>
</tr>
<tr>
<td>Drosophila melanogaster</td>
<td>21973</td>
<td>13790</td>
</tr>
<tr>
<td>Danio rerio</td>
<td>46848</td>
<td>25706</td>
</tr>
<tr>
<td>Xenopus laevis</td>
<td>44571</td>
<td>43235</td>
</tr>
</tbody>
</table>
<blockquote>
<p>All known human protein-coding genes are expertly  curated and described in 20 197 UniProtKB/Swiss-Prot entries (numbers in this publication are from UniProt re-lease 2015_10).</p>
</blockquote>
<blockquote>
<p>A complementary pipeline for the import of predicted human protein sequences in UniProtKB/TrEMBL has been developed in collaboration with Ensembl to complete the set of human isoform sequences.</p>
</blockquote>

          
        
      
    </div>
    
    
    

    

    

    

    <footer class="post-footer">
      

      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </div>
  
  
  
  </article>


    
  </section>

  
  <nav class="pagination">
    <a class="extend prev" rel="prev" href="/"><i class="fa fa-angle-left"></i></a><a class="page-number" href="/">1</a><span class="page-number current">2</span><a class="page-number" href="/page/3/">3</a><a class="extend next" rel="next" href="/page/3/"><i class="fa fa-angle-right"></i></a>
  </nav>



          </div>
          


          

        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    
    <div class="sidebar-inner">

      

      

      <section class="site-overview-wrap sidebar-panel sidebar-panel-active">
        <div class="site-overview">
          <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
            
              <img class="site-author-image" itemprop="image"
                src="https://timgsa.baidu.com/timg?image&quality=80&size=b9999_10000&sec=1534345547319&di=b9e880681b19e0cefb3132f38ccb0281&imgtype=0&src=http%3A%2F%2Fimg.duoziwang.com%2F2017%2F09%2F1607450267817.jpg"
                alt="HaolinZhong" />
            
              <p class="site-author-name" itemprop="name">HaolinZhong</p>
              <p class="site-description motion-element" itemprop="description">find a way</p>
          </div>

          <nav class="site-state motion-element">

            
              <div class="site-state-item site-state-posts">
              
                <a href="/archives/">
              
                  <span class="site-state-item-count">14</span>
                  <span class="site-state-item-name">日志</span>
                </a>
              </div>
            

            

            

          </nav>

          

          

          
          

          
          

          

        </div>
      </section>

      

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright">&copy; <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-user"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">HaolinZhong</span>

  
</div>


  <div class="powered-by">由 <a class="theme-link" target="_blank" href="https://hexo.io">Hexo</a> 强力驱动</div>



  <span class="post-meta-divider">|</span>



  <div class="theme-info">主题 &mdash; <a class="theme-link" target="_blank" href="https://github.com/iissnan/hexo-theme-next">NexT.Muse</a> v5.1.4</div>




        







        
      </div>
    </footer>

    
      <div class="back-to-top">
        <i class="fa fa-arrow-up"></i>
        
      </div>
    

    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  












  
  
    <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>
  

  
  
    <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>
  

  
  
    <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>
  

  
  
    <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>
  

  
  
    <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>
  

  
  
    <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>
  


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.4"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.4"></script>



  
  

  

  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.4"></script>



  


  




	





  





  












  





  

  

  

  
  

  

  

  

</body>
</html>
